File: sdk/csharp/src/LiteGraph.Sdk/UserMaster.cs
namespace LiteGraph.Sdk
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;

    /// <summary>
    /// User.
    /// </summary>
    public class UserMaster
    {
        #region Public-Members

        /// <summary>
        /// GUID.
        /// </summary>
        public Guid GUID { get; set; } = Guid.NewGuid();

        /// <summary>
        /// Tenant GUID.
        /// </summary>
        public Guid TenantGUID { get; set; } = Guid.NewGuid();

        /// <summary>
        /// First name.
        /// </summary>
        public string FirstName { get; set; } = string.Empty;

        /// <summary>
        /// Last name.
        /// </summary>
        public string LastName { get; set; } = string.Empty;

        /// <summary>
        /// Email.
        /// </summary>
        public string Email { get; set; } = string.Empty;

        /// <summary>
        /// Password.
        /// </summary>
        public string Password { get; set; } = string.Empty;

        /// <summary>
        /// Active.
        /// </summary>
        public bool Active { get; set; } = true;

        /// <summary>
        /// Indicates whether the user is a system administrator (server-wide superuser).  Default is false.
        /// </summary>
        public bool IsSystemAdmin { get; set; } = false;

        /// <summary>
        /// Indicates whether the user is a tenant administrator within their own tenant.  Default is false.
        /// </summary>
        public bool IsTenantAdmin { get; set; } = false;

        /// <summary>
        /// Creation time, in UTC.
        /// </summary>
        public DateTime CreatedUtc { get; set; } = DateTime.UtcNow;

        /// <summary>
        /// Timestamp from last update, in UTC.
        /// </summary>
        public DateTime LastUpdateUtc { get; set; } = DateTime.UtcNow;

        #endregion

        #region Private-Members

        #endregion

        #region Constructors-and-Factories

        /// <summary>
        /// Instantiate.
        /// </summary>
        public UserMaster()
        {

        }

        #endregion

        #region Public-Methods

        #endregion

        #region Private-Methods

        #endregion
    }
}

File: sdk/python/src/litegraph_sdk/models/user_master.py
from datetime import datetime, timezone
from pydantic import BaseModel, Field, ConfigDict


class UserMasterModel(BaseModel):
    """User master model."""

    guid: str = Field(default="", alias="GUID")
    tenant_guid: str = Field(default="", alias="TenantGUID")
    first_name: str = Field(default="", alias="FirstName")
    last_name: str = Field(default="", alias="LastName")
    email: str = Field(default="", alias="Email")
    password: str = Field(default="", alias="Password")
    active: bool = Field(default=True, alias="Active")
    is_system_admin: bool = Field(default=False, alias="IsSystemAdmin")
    is_tenant_admin: bool = Field(default=False, alias="IsTenantAdmin")
    created_utc: datetime = Field(default_factory=lambda: datetime.now(timezone.utc), alias="CreatedUtc")
    last_update_utc: datetime = Field(default_factory=lambda: datetime.now(timezone.utc), alias="LastUpdateUtc")

    model_config = ConfigDict(populate_by_name=True)
